### PR TITLE
feat: Added a sequential indicies sampler

### PR DESCRIPTION
**Pull request by lakshjaisinghani**
A new sampler that takes in a sequence of indices and samples those from a dataset. This is useful when concurrent testing whilst building models. You can:

1. Select particular indices to be sampled.
2. Randomise them (using a manual seed) and select them all.

Ideally, this should be a part of a class `SubsetSampler` where we can define a flag `random = bool` where we can use the functionality of `SubsetRandomSampler` if random is set to true.

### CODE REVIEW

**Comment by ejguan** [comment]
Hey @lakshjaisinghani, thanks for taking interest in contributing PyTorch.

I don't think `SequentialIndicesSampler` is necessary as it's simply an iterable. It's same as directly providing `indices` to DataLoader.

**Comment by lakshjaisinghani** [comment]
Ohh yes, I see that `torch.utils.data.Subset(dataset, indices)` does the same thing, it's just not a sampler haha! 

Thanks @ejguan!
